68 spa st strong ip model

VaradK62442 opened the pull request.

VaradK62442 committed:
small fixes

VaradK62442 committed:
add len behaviour

VaradK62442 committed:
create file reader with data struct more suited to IP solvere

VaradK62442 committed:
add spa-st instance generator (random)

VaradK62442 committed:
initial implementation, need to squash bugs

VaradK62442 committed:
small bug fixes, check for infeasibility

BenOnTheBoard committed:
Projected list ordering bugfix

BenOnTheBoard committed:
Projected list ordering bugfix

VaradK62442 committed:
generate smaller instances

VaradK62442 committed:
minor qol tweaks

VaradK62442 committed:
checking if project in student pref list is funky since EPI is funky, check if project is in the keys instead

VaradK62442 committed:
checking if project in student pref list is funky since EPI is funky, check if project is in the keys instead

VaradK62442 committed:
Merge branch '68-spa-st-strong-ip-model' of github.com:VaradK62442/algmatch into 68-spa-st-strong-ip-model

VaradK62442 committed:
better displaying results

VaradK62442 committed:
make constraint clearer

VaradK62442 committed:
ignore some more files txt, ilp, png

BenOnTheBoard committed:
toml numpy version typo

BenOnTheBoard committed:
slashes causing syntax warnings

BenOnTheBoard committed:
Total Unimodulatity Checker

BenOnTheBoard committed:
Renaming generator class for clarity

BenOnTheBoard committed:
Full TUM testing program

VaradK62442 committed:
make omega clearer

VaradK62442 committed:
better student project lecturer naming (-> s_i p_j l_k)

VaradK62442 committed:
better student project lecturer naming (-> s_i p_j l_k)

BenOnTheBoard committed:
SPASTAbstract Setup and check strong stability framework

BenOnTheBoard committed:
SPAST Strong Type 1 definitions

BenOnTheBoard committed:
SPAST Strong Type 2 Definitions

BenOnTheBoard committed:
Bugs in Checking SPAST Strong Stability

Mostly to do with the different strucutre of the lecturer dictionary to include lkj, meaning the projects don't have their own rankings.

BenOnTheBoard committed:
SPASAbstract name typo

BenOnTheBoard committed:
Tie Unpacking for check strong stability

BenOnTheBoard committed:
SPAST Brute Force Class

BenOnTheBoard committed:
IPModel answers to dictionary

For comparison with bruteforcer.

BenOnTheBoard committed:
No solution bugfix and testing single random instances

BenOnTheBoard committed:
SPAST Strong IP Model testing

BenOnTheBoard committed:
Absolute import needed

BenOnTheBoard commented on src/algmatch/stableMatchings/studentProjectAllocation/ties/entityPreferenceInstance.py:
```diff
@@ -42,10 +43,16 @@ def __hash__(self) -> int:
+        return len(self.values) if self.isTie else 1
 
     def _remove_from_tied(self, value):
         if isinstance(self.values, tuple): 
```
Is the use of isinstance in _remove_from_tied still necessary or does it need to be refactored with self.isTie?

BenOnTheBoard reviewed (commented):
All LGTM, just double check the method I pointed out in EntityPreferenceInstance and any related methods.

BenOnTheBoard commented:
Also: I introduce SPASTAbstract and SPASSolver and SPASSuper may need refactoring on this basis.
SPASTSuper has an incomplete check stability method which can be moved to the new _check_super_stability in SPASTAbstract and completed there.

BenOnTheBoard committed:
Rearranging methods for readability

Collecting occupancy methods which puts alpha next to beta where it's meant to be.

VaradK62442 committed:
review changes - use self.isTie instead of isinstance

VaradK62442 commented:
happy to merge if you are

regarding above comment, agree, can be made into an issue?
